{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "c4832388-12db-46b2-8730-026a3673e78e",
   "metadata": {},
   "outputs": [],
   "source": [
    "# import configparser\n",
    "# try:\n",
    "#    config = open(\"config.ini\")\n",
    "# except FileNotFoundError:\n",
    "#    API = input(\"Please Enter Open AI API:\")\n",
    "# else: \n",
    "#    pass"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d16f36b5-8bb9-49b8-be19-b70c04b83832",
   "metadata": {},
   "source": [
    "## API test"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "7ab6f992-6039-444d-843e-511090d46f8e",
   "metadata": {},
   "outputs": [],
   "source": [
    "import openai\n",
    "import re\n",
    "openai.api_key = \"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\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "12a1006d-e942-41e7-a1c2-1866b616186a",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "8ebbc194-1540-4b70-816b-73f89dbbe3cb",
   "metadata": {},
   "outputs": [],
   "source": [
    "def ask_abs(text,question=\"\",usage=False):\n",
    "   Require = \"Please answer the questions below without complete sentence refer to this artical\"\n",
    "   default_Qabs = '''\n",
    "         1. How many new strains have been discovered?\\n\n",
    "         2. What's the name of those new strains?\\n\n",
    "         3. What lineage does this species belong to?\\n\n",
    "         4. What the characteristics of these strains?\\n\n",
    "         5. What its best optimal growth environment?\\n\n",
    "         6. What is the metabolic profile of the strain?\\n\"\n",
    "   '''\n",
    "   if not question:\n",
    "      question = default_Qabs\n",
    "   Prompt = \"{Require}:\\n{Abstract}\\nQuestions:\\n{Question}\".format(Require = Require, Abstract = test_text, Question = question)\n",
    "   response = openai.Completion.create(\n",
    "     model=\"text-davinci-003\",\n",
    "     prompt=Prompt,\n",
    "     temperature=0,\n",
    "     max_tokens=200,\n",
    "     top_p=1,\n",
    "     frequency_penalty=0,\n",
    "     presence_penalty=0\n",
    "   )\n",
    "   Answer = response['choices'][0]['text']\n",
    "   pattern = r\"\\d+\\.\\s+(.*)\"\n",
    "   Answer_list = re.findall(pattern,Answer)\n",
    "   if usage:\n",
    "      print(f'{response[\"usage\"][\"prompt_tokens\"]} prompt tokens used.')\n",
    "   return Answer_list\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "e6ee71d7-7236-4afb-a035-4bce9329b8c4",
   "metadata": {},
   "outputs": [],
   "source": [
    "test_text = '''\n",
    "A novel bacterial strain, designated BS-T2-15T\n",
    ", isolated from forest soil in close proximity to decaying oak wood, was charac\u0002terized using a polyphasic taxonomic approach. Phylogenetic analyses based on 16S rRNA gene sequences as well as phy\u0002logenomic analyses based on coding sequences of 340 concatenated core proteins indicated that strain BS-T2-15T\n",
    " forms a \n",
    "distinct and robust lineage in the Rubrivivax–Roseateles–Leptothrix–Azohydromonas–Aquincola–Ideonella branch of the order \n",
    "Burkholderiales. The amino acid identity and the percentage of conserved proteins between the genome of strain BS-T2-15T\n",
    "and genomes of closely related type strains ranged from 64.27 to 66.57% and from 40.89 to 49.27%, respectively, providing \n",
    "genomic evidence that strain BS-T2-15T\n",
    " represents a new genus. Its cells are Gram-stain-negative, aerobic, motile by a polar \n",
    "flagellum, rod-shaped and form incrusted white to ivory colonies. Optimal growth is observed at 20–22°C, pH 6 and 0% NaCl. \n",
    "The predominant fatty acids of strain BS-T2-15T\n",
    " are C16:1ω7c, C16:0\n",
    " and C14:0\n",
    " 2-OH. Its polar lipid profile consists of a mixture of \n",
    "phosphatidylethanolamine, diphosphatidylglycerol and phosphatidylglycerol and its main respiratory quinone is ubiquinone 8. \n",
    "The estimated size of its genome is 6.28Mb with a DNA G+C content of 69.56mol%. Therefore, on the basis of phenotypic and \n",
    "genotypic properties, the new strain BS-T2-15T\n",
    " represents a novel genus and species for which the name Scleromatobacter \n",
    "humisilvae gen. nov., sp. nov., is proposed. The type strain is BS-T2-15T\n",
    " (DSM 113115T\n",
    "=UBOCC-M-3373T\n",
    ").\n",
    "'''\n",
    "test_text2 = '''\n",
    "A Gram-stain-positive, rod-shaped, non-spore-forming, catalase-negative, urease-negative, homofermentative and facultatively anaerobic strain, named WILCCON 0076T, was isolated from a wild ferment of pieces of a ‘Kampung’ durian fruit collected on the island of Ubin (Pulau Ubin), Singapore. The durian had fallen to the ground from a durian tree (Durio zibethinus), on which a group of long-tailed macaques had been observed picking and eating the fruits. Comparative analyses of 16S rRNA gene sequences indicated that WILCCON 0076T potentially represented a novel species within the genus Ligilactobacillus, with the most closely related type strain being Ligilactobacillus agilis DSM 20509T (16S rRNA gene sequence similarity of 97.2 %). Average nucleotide identity and digital DNA–DNA hybridization prediction values were only 86.0% and 18.9 %, respectively. On the basis of the results of a polyphasic approach that included phylogenomic, chemotaxonomic and morphological analyses, we propose a novel species with the name Ligilactobacillus ubinensis sp. nov. (type strain WILCCON 0076T=DSM 114293T=LMG 32698T).\n",
    "'''"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "bb97c4c7-dc0c-49da-8beb-745f4ff60500",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['One',\n",
       " 'Scleromatobacter humisilvae',\n",
       " 'The Rubrivivax–Roseateles–Leptothrix–Azohydromonas–Aquincola–Ideonella branch of the order Burkholderiales',\n",
       " 'Gram-stain-negative, aerobic, motile by a polar flagellum, rod-shaped and form incrusted white to ivory colonies',\n",
       " '20–22°C, pH 6 and 0% NaCl',\n",
       " 'Predominant fatty acids of C16:1ω7c, C16:0 and C14:0 2-OH; polar lipid profile consists of a mixture of phosphatidylethanolamine, diphosphatidylglycerol and phosphatidylglycerol; main respiratory quinone is ubiquinone 8.']"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ask_abs(test_text2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "177ee6b6-72d6-4c12-8d7f-d8cd72543f8a",
   "metadata": {},
   "outputs": [],
   "source": [
    "import requests\n",
    "from bs4 import BeautifulSoup\n",
    "\n",
    "url = 'https://www.microbiologyresearch.org/content/journal/ijsem/10.1099/ijsem.0.005733'\n",
    "#Send HTTP request\n",
    "headers = {\n",
    "    'User-Agent':\n",
    "    'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/79.0.3945.88 Safari/537.36',\n",
    "}\n",
    "\n",
    "url_head = \"https://www.microbiologyresearch.org/\"\n",
    "response = requests.get(url,headers)\n",
    "match = re.search(r'data-fullTexturl=\"(.*?)\"',response.text)\n",
    "if match:\n",
    "    url_adhere = match.group(1)\n",
    "redirect = url_head + url_adhere"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "fa18422b-66b0-4136-844a-f8df38e26839",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Parsing full-text content\n",
    "full_text_url = requests.get(redirect,headers).text\n",
    "full_text = requests.get(full_text_url,headers)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "d9448efc-2120-4290-a37f-4d4dfc834606",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\ProgramData\\Miniconda3\\envs\\bim\\lib\\site-packages\\bs4\\builder\\__init__.py:545: XMLParsedAsHTMLWarning: It looks like you're parsing an XML document using an HTML parser. If this really is an HTML document (maybe it's XHTML?), you can ignore or filter this warning. If it's XML, you should know that using an XML parser will be more reliable. To parse this document as XML, make sure you have the lxml package installed, and pass the keyword argument `features=\"xml\"` into the BeautifulSoup constructor.\n",
      "  warnings.warn(\n"
     ]
    }
   ],
   "source": [
    "soup = BeautifulSoup(full_text.text, 'html.parser')\n",
    "elements_to_remove = soup.find_all('div', {'class': 'dropDownMenu'})\n",
    "for element in elements_to_remove:\n",
    "    element.decompose()\n",
    "elements_to_remove = soup.find_all('div', {'class': 'menuButton'})\n",
    "for element in elements_to_remove:\n",
    "    element.decompose()\n",
    "\n",
    "# 查找需要的内容\n",
    "title = soup.find('h1').text.strip()\n",
    "authors = [a.text.strip() for a in soup.find_all('span')]\n",
    "article = soup.find_all('div', {'class': 'articleSection'})\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "5d14215a-645d-40f2-becf-1cef6e251a0d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "b'\\n                                \\xc3\\x82\\xc2\\xa0\\n                            \\xc3\\x82\\xc2\\xa0Ligilactobacillus ubinensis sp. nov., a novel species isolated from the wild ferment of a durian fruit (Durio zibethinus)Yu Chyuan Heng1; Nandita Menon1; Binbin Chen1; Bryan Zong Lin Loo1; Garrett Wei Jie Wong1; Amber Ching Han Lim1; Shaktheeshwari Silvaraju1; Sandra Kittelmann1*1\\xc3\\xa2\\xc2\\x80\\xc2\\x8bWilmar International Limited, WIL@NUS Corporate Laboratory, Centre for Translational Medicine, National University of Singapore, Singapore, Singapore*Correspondence: Sandra Kittelmann, sandra.kittelmann@sg.wilmar-intl.comFN1Subcultures of WILCCON 0076T have been deposited in the Leibniz Institute DSMZ \\xc3\\xa2\\xc2\\x80\\xc2\\x93 German Collection of Microorganisms and Cell Cultures GmbH and Belgian Coordinated Collections of Microorganisms/Laboratory for Microbiology of the Faculty of Sciences of the Ghent University (BCCM/LMG) Bacteria Collection under the accession numbers DSM 114293T and LMG 32698T, respectively. The 16S rRNA gene sequence, the whole genomeshotgun project, and the draft genome assembly of strain WILCCON 0076T have been deposited in GenBank/ENA/DDBJ under the accession numbers OK166812, JAIULA010000000 and GCA_024160875, respectively.FN2Three supplementary figures and three supplementary tables are available with the online version of this article.\\xc3\\x82\\xc2\\xa0'\n",
      "b'\\n                                \\xc3\\x82\\xc2\\xa0\\n                            \\xc3\\x82\\xc2\\xa0'\n",
      "b'ABSTRACT\\xc3\\x82\\xc2\\xa0A Gram-stain-positive, rod-shaped, non-spore-forming, catalase-negative, urease-negative, homofermentative and facultatively anaerobic strain, named WILCCON 0076T, was isolated from a wild ferment of pieces of a \\xc3\\xa2\\xc2\\x80\\xc2\\x98Kampung\\xc3\\xa2\\xc2\\x80\\xc2\\x99 durian fruit collected on the island of Ubin (Pulau Ubin), Singapore. The durian had fallen to the ground from a durian tree (Durio zibethinus), on which a group of long-tailed macaques had been observed picking and eating the fruits. Comparative analyses of 16S rRNA gene sequences indicated that WILCCON 0076T potentially represented a novel species within the genus Ligilactobacillus, with the most closely related type strain being Ligilactobacillus agilis DSM 20509T (16S rRNA gene sequence similarity of 97.2\\xc3\\xa2\\xc2\\x80\\xc2\\x8a%). Average nucleotide identity and digital DNA\\xc3\\xa2\\xc2\\x80\\xc2\\x93DNA hybridization prediction values were only 86.0% and 18.9\\xc3\\xa2\\xc2\\x80\\xc2\\x8a%, respectively. On the basis of the results of a polyphasic approach that included phylogenomic, chemotaxonomic and morphological analyses, we propose a novel species with the name Ligilactobacillus ubinensis sp. nov. (type strain WILCCON 0076T=DSM 114293T=LMG 32698T).\\nKeywordsDurio zibethinus, fermented durian, lactobacilli, Ligilactobacillus ubinensis sp. nov, Singapore\\xc3\\x82\\xc2\\xa0'\n",
      "b'ABSTRACT\\xc3\\x82\\xc2\\xa0'\n",
      "b'Abbreviations\\xc3\\x82\\xc2\\xa0AAI, average amino acid identity; ANI, average nucleotide identity; cAAI, core amino acid identity; CNI, core nucleotide identity; dDDH, digital DNA-DNA hybridization; Dpm, diaminopimelic acid; GSS, gene sequence similarity; HMM, Hidden Markov models; SCGs, single-copy core genes.\\n\\xc3\\x82\\xc2\\xa0'\n",
      "b'Abbreviations\\xc3\\x82\\xc2\\xa0'\n",
      "b'Introduction\\xc3\\x82\\xc2\\xa0The \\xc3\\xa2\\xc2\\x80\\xc2\\x98lactobacilli\\xc3\\xa2\\xc2\\x80\\xc2\\x99 as re-defined by Zheng and co-workers in 2020, encompass the three former genera Lactobacillus (L.), Paralactobacillus and Pediococcus [1]. This group of Gram-positive bacteria within the phylum Bacillota (previously Firmicutes), family Lactobacillaceae is a rapidly growing phylogenetic lineage. To date, it accommodates just over 300 species and subspecies with validly published names, which were recently re-classified into 26 different genera [1]. Members of the lactobacilli are adapted to various plant and host-associated niches [2]. Some have played important roles, both economically and culturally, in fermented food and beverage production. Their use also extends into other areas of biotechnology, e.g. in bioprotection or the bioproduction of value-added compounds.\\nThe genus Ligilactobacillus (Lg.) is most closely related to two other genera of lactobacilli, namely Liquorilactobacillus (Lq.) and Dellaglioa (D.). It comprises a group of predominantly host-adapted, homofermentative and vancomycin-resistant species [1, 3]. At the time of writing, 16 species of the genus Ligilactobacillus have been recognized, with Ligilactobacillus salivarius representing the type species. In contrast to the vast majority of lactobacilli, many members of the genus Ligilactobacillus are motile through the development and activity of a flagellum [4, 5]. Free-living flagellated species may have a competitive advantage over non-motile species with respect to adaptation to environmental changes, nutrient acquisition, biofilm formation and adhesion [4]. Inside the host gastrointestinal tract, the globular protein flagellin, which forms the flagellar filament, is thought to act as a ligand for activating signalling pathways and modulating host immune cells [6]. Despite the recent re-classification, genomic data indicates that the genus Ligilactobacillus is still heterogeneous [1]. It has been proposed that the heterogeneity within this genus stems from the specialized adaptations of different clades to different ecological niches, e.g. through the maintenance of megaplasmids and specific exopolysaccharide gene clusters [7].\\nHere we describe strain WILCCON 0076T, which was isolated from the wild ferment of pieces of the husk and the flesh of a durian fruit (Durio zibethinus) collected in Singapore. The strain is most closely related to the type strains of Ligilactobacillus agilis and Ligilactobacillus equi, but presents substantial genomic, chemotaxonomic and phenotypic differences. We propose to establish the species name Ligilactobacillus ubinensis sp. nov. for this strain.\\n\\xc3\\x82\\xc2\\xa0'\n",
      "b'Introduction\\xc3\\x82\\xc2\\xa0'\n",
      "b'Isolation and ecology\\xc3\\x82\\xc2\\xa0Fermented durian paste, or \\xc3\\xa2\\xc2\\x80\\xc2\\x98tempoyak\\xc3\\xa2\\xc2\\x80\\xc2\\x99, is a traditional condiment that is widely consumed in Southeast Asia, mainly in Indonesia and Malaysia. For a trial to isolate useful microorganisms for fruit and vegetable fermentations, parts of a durian fruit were collected from the ground next to a durian tree on the island of Ubin (Pulau Ubin), Singapore (latitude 1.4126\\xc3\\x82\\xc2\\xb0 N, longitude 103.9577\\xc3\\x82\\xc2\\xb0 E). It is of note that, on the same tree, a group of long-tailed macaques were observed picking durian fruits, feeding on them and occasionally dropping intact unripe fruits to the ground. In the laboratory, durian cuttings, consisting of both the husk and the flesh, were transferred into a glass jar, topped with sterile water and, apart from occasional ventilation, left to ferment at 20\\xc3\\xa2\\xc2\\x80\\xc2\\x89\\xc3\\x82\\xc2\\xb0C inside a Coy anoxic chamber (atmosphere 4\\xc3\\xa2\\xc2\\x80\\xc2\\x8a% hydrogen, 5\\xc3\\xa2\\xc2\\x80\\xc2\\x8a% carbon dioxide and 91\\xc3\\xa2\\xc2\\x80\\xc2\\x8a% nitrogen; Coy Laboratory Products). After 14\\xc3\\xa2\\xc2\\x80\\xc2\\x89days, a dilution series of the sample was spread onto de Man Rogosa Sharpe (MRS; BD Difco) agar plates and incubated for 2\\xc3\\xa2\\xc2\\x80\\xc2\\x89days at 37\\xc3\\xa2\\xc2\\x80\\xc2\\x89\\xc3\\x82\\xc2\\xb0C under anoxic conditions. Single colonies of WILCCON 0076T were isolated (Fig. 1a) and identified as representing a potentially novel species of the genus Ligilactobacillus. Here, we report the genotypic and phenotypic characterization of WILCCON 0076T.\\nFig. 1.Images of (a) colonies and (b) Gram-stained cells (captured using light microscopy) of WILCCON 0076T. Colonies and cells were cultivated for 2\\xc3\\xa2\\xc2\\x80\\xc2\\x89days in MRS medium. The scale bar in panel (b) represents 10 \\xc3\\x82\\xc2\\xb5m.\\nFig. 1.Click to view\\xc3\\x82\\xc2\\xa0\\xc3\\x82\\xc2\\xa0'\n",
      "b'Isolation and ecology\\xc3\\x82\\xc2\\xa0'\n",
      "b'16S rRNA phylogeny and genome features\\xc3\\x82\\xc2\\xa0Genomic DNA was extracted from a pure culture of WILCCON 0076T (derived from a single isolated colony on an MRS plate) using the Maxwell 16 FFS nucleic acid extraction system (Promega) and paired-end sequenced using an Illumina HiSeq system (2\\xc3\\x83\\xc2\\x97150\\xc3\\xa2\\xc2\\x80\\xc2\\x89bp reads; NovogeneAIT Genomics Singapore). The 10\\xc3\\xa2\\xc2\\x80\\xc2\\x8a761\\xc3\\xa2\\xc2\\x80\\xc2\\x8a610 raw reads obtained were further quality trimmed and filtered using Trimmomatic (version 0.39; https://github.com/usadellab/Trimmomatic) [8] to yield 10\\xc3\\xa2\\xc2\\x80\\xc2\\x8a639\\xc3\\xa2\\xc2\\x80\\xc2\\x8a052 clean reads. Subsequent de novo assembly with SPAdes (version 3.15.2; https://github.com/ablab/spades) [9] generated a 2589\\xc3\\xa2\\xc2\\x80\\xc2\\x8a491\\xc3\\xa2\\xc2\\x80\\xc2\\x89bp draft genome (GenBank/ENA/DDBJ accession number GCA_024160875) comprising of 86 scaffolds. The draft genome was assessed for completeness and contamination using CheckM (version 1.1.3; https://github.com/Ecogenomics/CheckM) [10] before annotation using Prokka (version 1.14.6; https://github.com/tseemann/prokka) [11]. The protein annotations were later searched against the KEGG database (version 98.1; https://www.genome.jp/kegg/) [12] for metabolic pathway inference and reconstruction. ABRicate (version 1.0.1; https://github.com/tseemann/abricate) was used to screen for antimicrobial resistance and virulence genes. A summary of the genome characteristics of WILCCON 0076T is shown in Table 1.\\nTable 1.Genome characteristics of WILCCON 0076T\\nThe two partial, non-overlapping 16S rRNA gene sequences extracted from the draft genome (scaffolds 48 and 55) using Barrnap (version 0.9; https://github.com/tseemann/barrnap) were assembled into an almost full-length sequence (1\\xc3\\xa2\\xc2\\x80\\xc2\\x8a522\\xc3\\xa2\\xc2\\x80\\xc2\\x89bp; GenBank/ENA/DDBJ accession number OK166812) upon verification with the sequence obtained by polymerase chain reaction amplification using the bacterial universal primer set (27F and 1492R). The results of a preliminary search of the 16S rRNA gene sequence using BLASTn against the non-redundant nucleotide collection database from the National Centre for Biotechnology Information [13] indicated that WILCCON 0076T might represent a novel taxon within the family Lactobacillaceae. Pairwise 16S rRNA gene sequence similarity (GSS) values with related taxa were calculated using the method recommended by Meier-Kolthoff and co-workers [14] through the Genome-to-Genome Calculator\\xc3\\xa2\\xc2\\x80\\xc2\\x99s Phylogeny Service (http://ggdc.dsmz.de/phylogeny-service.php#) [15]. To infer a 16S rRNA gene sequence phylogeny, a multiple sequence alignment was generated using the online muscle tool [16] of the European Molecular Biology Laboratory\\xc3\\xa2\\xc2\\x80\\xc2\\x93European Bioinformatics Institute (EMBL\\xc3\\xa2\\xc2\\x80\\xc2\\x93EBI; https://www.ebi.ac.uk/Tools/msa/muscle/) [17]. A maximum-likelihood tree was reconstructed using IQ-TREE (version 2.1.2; http://www.iqtree.org/) [18], based on the best TVMe+I+G4\\xc3\\xa2\\xc2\\x80\\xc2\\x89DNA substitution model identified using the \\xc3\\xa2\\xc2\\x80\\xc2\\x98ModelFinder\\xc3\\xa2\\xc2\\x80\\xc2\\x99 method [19] and supported with 1000 ultrafast bootstrapping and Shimodaira\\xc3\\xa2\\xc2\\x80\\xc2\\x93Hasegawa-like approximate likelihood ratio tests (SH-aLRT) [20]. FigTree (version 1.4.4; https://github.com/rambaut/figtree) was used for graphical visualisation of the resulting tree. To estimate the whole-genome relatedness of WILCCON 0076T with related taxa, pairwise average nucleotide identity (ANI), average amino acid identity (AAI) and digital DNA\\xc3\\xa2\\xc2\\x80\\xc2\\x93DNA hybridization (dDDH) prediction values were calculated using pyani (version 0.2.11; https://github.com/widdowquinn/pyani) [21], CompareM (version 0.1.2; https://github.com/dparks1134/CompareM) and Formula 2 of Genome-to-Genome Distance Calculator (version 3.0; http://ggdc.dsmz.de/ggdc.php) [15], respectively.\\nTo evaluate the core-genome relatedness of WILCCON 0076T with related taxa, single-copy core genes (SCGs) were extracted using the SCARAP pipeline (version 0.3.1; https://github.com/SWittouck/SCARAP) [22]. For this, OrthoFinder (version 2.5.2; https://github.com/davidemms/OrthoFinder) [23] was initially used to conduct a full gene family clustering to identify conserved orthogroups that were present in the genomes of 31 type strains of type species of the family Lactobacillaceae (seed genomes). Hidden Markov models (HMM) of these orthogroups were subsequently curated and used to mine for candidate SCGs in all query genomes using HMMER (version 3.3.2; http://hmmer.org/) [24]. Final filtering, which enforced a strict 100\\xc3\\xa2\\xc2\\x80\\xc2\\x8a% single-copy presence in all query genomes, identified 255 definitive SCGs from the initial 365 candidate SCGs. To reconstruct a phylogenomic tree, each of the 255 SCGs were aligned at the amino acid level using MAFFT (version 7.271; https://github.com/GSLBiotech/mafft) [25] before concatenation into a large protein alignment using the \\xc3\\xa2\\xc2\\x80\\xc2\\x98supermatrix\\xc3\\xa2\\xc2\\x80\\xc2\\x99 method of SCARAP. Poorly aligned or gapped positions of the alignment were removed using the \\xc3\\xa2\\xc2\\x80\\xc2\\x98automated1\\xc3\\xa2\\xc2\\x80\\xc2\\x99 method of trimAl (version 1.4; https://vicfero.github.io/trimal/) [26] optimized for maximum likelihood phylogenomic tree reconstruction. A phylogeny was inferred with IQ-TREE [18], using the best LG+F+R6 protein substitution model identified by the \\xc3\\xa2\\xc2\\x80\\xc2\\x98ModelFinder\\xc3\\xa2\\xc2\\x80\\xc2\\x99 method [19] and supported with 1000 ultrafast bootstrapping and SH-aLRT tests [20]. The resulting phylogenomic tree was visualised using FigTree. Similarly, to calculate pairwise core nucleotide identity (CNI) prediction values, the \\xc3\\xa2\\xc2\\x80\\xc2\\x98supermatrix\\xc3\\xa2\\xc2\\x80\\xc2\\x99 method of SCARAP was used first to align each SCG at the nucleotide level using the corresponding protein alignment as a reference (reverse alignment) before concatenation into a large nucleotide alignment. Pairwise CNI prediction values were calculated by subtracting from one the pairwise nucleotide differences computed from the large concatenated nucleotide alignment using the distmat tool of wEMBOSS (version 6.6.0.0; https://www.bioinformatics.nl/cgi-bin/emboss/distmat) [27] without correcting for multiple substitutions [22]. Pairwise core amino acid identity (cAAI) prediction values were calculated with CompareM, using only the protein sequences of the 255 SCGs as the input.\\nWILCCON 0076T exhibited pairwise 16S rRNA gene sequence similarity values higher than the proposed 94.5\\xc3\\xa2\\xc2\\x80\\xc2\\x8a% genus-level cut-off threshold [1] with the type species Liquorilactobacillus mali DSM 20444T (96.3\\xc3\\xa2\\xc2\\x80\\xc2\\x8a%; M58824) and Lg. salivarius DSM 20555T (95.3\\xc3\\xa2\\xc2\\x80\\xc2\\x8a%; AF089108). This indicated that WILCCON 0076T could represent a member of either of the two genera (Fig. 2a and Table S1). The highest pairwise AAI prediction value, which was greater than the proposed 68.0\\xc3\\xa2\\xc2\\x80\\xc2\\x8a% genus-level cut-off threshold [1], was found with the type strain Lg. agilis DSM 20509T (72.6\\xc3\\xa2\\xc2\\x80\\xc2\\x8a%; GCA_001436215; Fig. 2a and Table 2). In addition, WILCCON 0076T clustered into a monophyletic clade with 11 type strains of species of the genus Ligilactobacillus (including the type species Lg. salivarius DSM 20555T) on the 16S rRNA gene sequence phylogenetic tree (Fig. 3). This genus assignment was further supported by the results of pairwise CNI and cAAI analyses, which indicated higher estimated mean and median values between WILCCON 0076T and the type strains of species of the genus Ligilactobacillus than with those of the genus Liquorilactobacillus (Fig. 2b). Significantly, the ranges of pairwise AAI, CNI and cAAI prediction values between WILCCON 0076T and the type strains of species of the genus Liquorilactobacillus fell into the inter-genus range of existing species of the genera Ligilactobacillus and Liquorilactobacillus (Fig. 2b). While Liquorilactobacillus has previously been described as an exclusive genus with intra-genus pairwise AAI and cAAI ranges greater than the inter-genus counterparts, Ligilactobacillus has been described as a non-exclusive genus with overlapping intra- and inter-genus pairwise AAI and cAAI ranges [1]. The latter matched the distribution of pairwise AAI, CNI and cAAI between WILCCON 0076T and the type strains of species of the genus Ligilactobacillus (Fig. 2b and Fig. S1, available in the online version of this article), which supported the assignment into the genus Ligilactobacillus. Further evidence was derived from the phylogenomic tree, in which WILCCON 0076T was found to form a distinct monophyletic clade with the type strains of all 15 species of the genus Ligilactobacillus (Fig. 4). The highest pairwise 16S rRNA GSS (97.2\\xc3\\xa2\\xc2\\x80\\xc2\\x8a%), ANI (86.0\\xc3\\xa2\\xc2\\x80\\xc2\\x8a%), dDDH (18.9\\xc3\\xa2\\xc2\\x80\\xc2\\x8a%) and CNI (72.3\\xc3\\xa2\\xc2\\x80\\xc2\\x8a%) prediction values were found with the type strain Lg. agilis DSM 20509T (Tables 2 and S2) and were all well below the generally accepted species-level thresholds of 98.65\\xc3\\xa2\\xc2\\x80\\xc2\\x8a%, 95.0\\xc3\\xa2\\xc2\\x80\\xc2\\x8a%, 70.0% and 94.0\\xc3\\xa2\\xc2\\x80\\xc2\\x8a%, respectively [22, 28]. The DNA G+C content of WILCCON 0076T was 33.7\\xc3\\xa2\\xc2\\x80\\xc2\\x89mol%, while that of the most closely related type strain Lg. agilis DSM 20509T was 43.9\\xc3\\xa2\\xc2\\x80\\xc2\\x89mol% [29], indicating that WILCCON 0076T is clearly distinct (more than 1.0\\xc3\\xa2\\xc2\\x80\\xc2\\x89mol% difference from the most closely related type strain). Altogether, these characteristics indicated that WILCCON 0076T represented a novel species within the genus Ligilactobacillus.\\nFig. 2.Pairwise 16S rRNA gene sequence similarity (GSS), average amino acid identity (AAI), core nucleotide identity (CNI) and core amino acid identity (cAAI) prediction values between WILCCON 0076T and closely related taxa of the genera Ligilactobacillus (Lg.), Liquorilactobacillus (Lq.) and Dellaglioa (D.). (a) Pairwise 16S rRNA GSS values greater than the proposed 94.5\\xc3\\xa2\\xc2\\x80\\xc2\\x8a% genus-level cut-off threshold [1] were found with the type species Lq. mali DSM 20444T and Lg. salivarius DSM 20555T; the highest pairwise AAI prediction value exceeding the proposed 68.0\\xc3\\xa2\\xc2\\x80\\xc2\\x8a% genus-level cut-off threshold [1] was found with the type strain Lg. agilis DSM 20509T. (b) The pairwise CNI and cAAI prediction values between WILCCON 0076T and all type strains of species of the genus Liquorilactobacillus and some type strains of species of the genus Ligilactobacillus were within the inter-genus range of existing species of the genera Ligilactobacillus and Liquorilactobacillus (grey-shaded area). Boxplots show the distributions of the pairwise CNI and cAAI prediction values. Mean and median values are represented by a cross (\\xc3\\x83\\xc2\\x97) and a line (-) within the boxes, respectively.\\nFig. 2.Click to view\\xc3\\x82\\xc2\\xa0Fig. 3.Phylogenetic tree based on 16S rRNA gene sequences showing the relationship of WILCCON 0076T (bold type) and closely related taxa of the genera Ligilactobacillus (Lg.), Liquorilactobacillus (Lq.) and Dellaglioa (D.). The maximum-likelihood tree was inferred under the TVMe+I+G4\\xc3\\xa2\\xc2\\x80\\xc2\\x89DNA substitution model (optimal log-likelihood of \\xc3\\xa2\\xc2\\x88\\xc2\\x927576.199) and rooted by midpoint-rooting. Lactobacillus (L.) delbrueckii subsp. delbrueckii DSM 20074T was used as an outgroup. SH-aLRT (left) and ultrafast bootstrap (right) values based on 1000 replications are indicated at branch points. Bar, 0.04 substitutions per nucleotide position.\\nFig. 3.Click to view\\xc3\\x82\\xc2\\xa0Fig. 4.Phylogenomic tree based on the concatenation of the multiple alignment of the protein sequences of 255 single-copy core genes (SCGs) showing the relationship of WILCCON 0076T (bold type) and closely related taxa of the genera Ligilactobacillus (Lg.), Liquorilactobacillus (Lq.) and Dellaglioa (D.). The maximum-likelihood tree was inferred under the LG+F+R6 protein substitution model (optimal log-likelihood of \\xc3\\xa2\\xc2\\x88\\xc2\\x921898007.037) and rooted by midpoint-rooting. Lactobacillus (L.) delbrueckii subsp. delbrueckii DSM 20074T was used as an outgroup. SH-aLRT (left) and ultrafast bootstrap (right) values based on 1000 replications are indicated at branch points. Bar, 0.09 substitutions per amino acid position.\\nFig. 4.Click to view\\xc3\\x82\\xc2\\xa0Table 2.Pairwise 16S rRNA gene sequence similarity (GSS), average amino acid identity (AAI), core nucleotide identity (CNI) and core amino acid identity (cAAI) prediction values between WILCCON 0076T and closely related taxa of the genera Ligilactobacillus (Lg.) and Liquorilactobacillus (Lq.). The type species of the two genera are highlighted in bold type\\n\\xc3\\x82\\xc2\\xa0'\n",
      "b'16S rRNA phylogeny and genome features\\xc3\\x82\\xc2\\xa0'\n",
      "b'Physiology and chemotaxonomy\\xc3\\x82\\xc2\\xa0Morphological and biochemical tests were performed after growth of WILCCON 0076T in MRS medium for 1\\xc3\\xa2\\xc2\\x80\\xc2\\x89day or on MRS agar plates for 2\\xc3\\xa2\\xc2\\x80\\xc2\\x89days at 30\\xc3\\xa2\\xc2\\x80\\xc2\\x89\\xc3\\x82\\xc2\\xb0C or 37\\xc3\\xa2\\xc2\\x80\\xc2\\x89\\xc3\\x82\\xc2\\xb0C anaerobically inside an anoxic chamber (atmosphere 4\\xc3\\xa2\\xc2\\x80\\xc2\\x8a% hydrogen, 5\\xc3\\xa2\\xc2\\x80\\xc2\\x8a% carbon dioxide and 91\\xc3\\xa2\\xc2\\x80\\xc2\\x8a% nitrogen; Coy Laboratory Products), unless stated otherwise. The Gram-stain-positive (Fig. 1b) and non-spore forming nature of this strain was confirmed by staining using a commercial Gram-staining kit (Millipore-Sigma) and the malachite green endospore stain (Sigma-Aldrich), respectively, followed by observation of stained cells under a biological microscope (BX53, Olympus). Haemolytic activity was checked on Columbia blood agar with 5\\xc3\\xa2\\xc2\\x80\\xc2\\x8a% sheep blood (Bio-Rad). After 3\\xc3\\xa2\\xc2\\x80\\xc2\\x89days, halo formation or green discolouration was not detected, indicating the absence of haemolytic activity. The absence of catalase activity was predicted from the absence of the catalase gene in the annotated draft genome and confirmed experimentally, by observation of the lack of gas production upon treatment with 3\\xc3\\xa2\\xc2\\x80\\xc2\\x8a% hydrogen peroxide. The d-/l-Lactic Acid Rapid Assay Kit (Megazyme) was used to determine the lactate isomers produced from the fermentation of glucose. Both d-lactate and l-lactate were produced by WILCCON 0076T at a ratio of 1\\xc3\\xa2\\xc2\\x80\\xc2\\x8a:\\xc3\\xa2\\xc2\\x80\\xc2\\x8a20, consistent with the presence of two putative d-lactate dehydrogenase genes (ldhA; LB941_03555 and LB941_05520) and two putative l-lactate dehydrogenase genes (ldh; LB941_01965 and LB941_05700) in the annotated draft genome. The absence of the ure operon, encoding the three subunits of the urease enzyme together with the accessory proteins, indicated the absence of urease activity in this strain. To confirm this, WILCCON 0076T was grown in MRS medium supplemented with 2\\xc3\\xa2\\xc2\\x80\\xc2\\x8a% (w/v) urea and 12 \\xc3\\x82\\xc2\\xb5g ml\\xc3\\xa2\\xc2\\x88\\xc2\\x921 phenol red. Degradation of urea is indicated by a colour change from yellow to pinkish-red, which was not observed for WILCCON 0076T. Vancomycin susceptibility was deduced computationally through confirmation of the absence of the van operon from the annotated draft genome.\\nThe optimal growth conditions for WILCCON 0076T were determined by measuring the cell densities (OD600) after cultivation at different temperatures (10, 20, 30, 37 and 45\\xc3\\xa2\\xc2\\x80\\xc2\\x89\\xc3\\x82\\xc2\\xb0C), pH values (3, 4, 5, 6, 7, 8, 9 and 10) and sodium chloride concentrations [2, 4, 6, 8, 10\\xc3\\xa2\\xc2\\x80\\xc2\\x8a% (w/v)] for 24\\xc3\\xa2\\xc2\\x80\\xc2\\x89h under anoxic conditions (Fig. S2). WILCCON 0076T grew optimally at both 30\\xc3\\xa2\\xc2\\x80\\xc2\\x89\\xc3\\x82\\xc2\\xb0C and 37\\xc3\\xa2\\xc2\\x80\\xc2\\x89\\xc3\\x82\\xc2\\xb0C. The optimal pH was 7 at both optimal temperatures. Cell growth was observed at 2\\xc3\\xa2\\xc2\\x80\\xc2\\x8a% but not at 4\\xc3\\xa2\\xc2\\x80\\xc2\\x8a% (w/v) sodium chloride concentration. Analyses for carbohydrate utilization, cellular fatty acid composition and peptidoglycan structure were carried out by DSMZ Services, Leibniz-Institut DSMZ - Deutsche Sammlung von Mikroorganismen und Zellkulturen GmbH, Braunschweig, Germany (https://www.dsmz.de/services/microorganisms/biochemical-analysis). The API 50 CHL system (bioM\\xc3\\x83\\xc2\\xa9rieux) was used to determine the carbohydrate utilization of WILCCON 0076T in comparison with the type strains of two closely related taxa; results are shown in Table 3. The cellular fatty acid composition of WILCCON 0076T was analysed by gas chromatography\\xc3\\xa2\\xc2\\x80\\xc2\\x93mass spectrometry of fatty acid methyl esters using the Sherlock Microbial Identification System (MIDI); results are summarised in Table 4. For cell-wall structural analysis, peptidoglycan composition was studied following total hydrolysis (4\\xc3\\xa2\\xc2\\x80\\xc2\\x89M hydrochloric acid at 100\\xc3\\xa2\\xc2\\x80\\xc2\\x89\\xc3\\x82\\xc2\\xb0C for 16\\xc3\\xa2\\xc2\\x80\\xc2\\x89h) and partial hydrolysis (4\\xc3\\xa2\\xc2\\x80\\xc2\\x89M hydrochloric acid at 100\\xc3\\xa2\\xc2\\x80\\xc2\\x89\\xc3\\x82\\xc2\\xb0C for 45\\xc3\\xa2\\xc2\\x80\\xc2\\x89min) using a protocol described previously [30]. The total hydrolysate of the peptidoglycan contained the amino sugar acid muramic acid and the amino acids diaminopimelic acid (Dpm), alanine (Ala) and glutamic acid (Glu). Further quantification of the amino acids by gas chromatography\\xc3\\xa2\\xc2\\x80\\xc2\\x93mass spectrometry of N-heptafluorobutyric amino acid isobutyl esters revealed a molar ratio of 0.9\\xc3\\xa2\\xc2\\x80\\xc2\\x8a:\\xc3\\xa2\\xc2\\x80\\xc2\\x8a1.0\\xc3\\xa2\\xc2\\x80\\xc2\\x8a:\\xc3\\xa2\\xc2\\x80\\xc2\\x8a1.0 between Dpm, Ala and Glu. The partial hydrolysate contained peptides M\\xc3\\xa2\\xc2\\x80\\xc2\\x93Ala, Ala\\xc3\\xa2\\xc2\\x80\\xc2\\x93Glu, M\\xc3\\xa2\\xc2\\x80\\xc2\\x93Ala\\xc3\\xa2\\xc2\\x80\\xc2\\x93Glu and Dpm\\xc3\\xa2\\xc2\\x80\\xc2\\x93Ala\\xc3\\xa2\\xc2\\x80\\xc2\\x93Dpm. The peptidoglycan type of WILCCON 0076T was concluded to be A1\\xc3\\x8e\\xc2\\xb3 meso-Dpm-direct (or type A31 according to Schumann [30]).\\nTable 3.Differential chemotaxonomic characteristics of WILCCON 0076T and closely related taxa of the genus Ligilactobacillus (Lg.). Strains: 1, WILCCON 0076T; 2, Lg. agilis DSM 20509T; 3, Lg. equi DSM 15833T. All strains produced acids from d-glucose, d-fructose, d-mannose, N-acetylglucosamine, arbutin, aesculin, salicin, maltose, lactose, melibiose, sucrose and raffinose. None of the strains produced acids from glycerol, erythritol, d-arabinose, l-xylose, d-adonitol, methyl \\xc3\\x8e\\xc2\\xb2-d-xylopyranoside, l-sorbose, dulcitol, inositol, methyl \\xc3\\x8e\\xc2\\xb1-d-mannopyranoside, methyl \\xc3\\x8e\\xc2\\xb1-d-glucopyranoside, inulin, melezitose, starch, glycogen, xylitol, turanose, d-lyxose, d-tagatose, d-fucose, l-fucose, d-arabitol, l-arabitol, potassium gluconate, potassium 2-ketogluconate and potassium 5-ketogluconate. All tested characteristics were determined under identical conditions in the present study. +, Positive; \\xc3\\xa2\\xc2\\x88\\xc2\\x92, negative; +/\\xc3\\xa2\\xc2\\x88\\xc2\\x92, weak reaction\\nTable 4.Cellular fatty acid composition of WILCCON 0076T. Values are percentages of total fatty acids. Fatty acids representing less than 0.5\\xc3\\xa2\\xc2\\x80\\xc2\\x8a% of the total fatty acids are not listed. The major components of cellular fatty acids are highlighted in bold type. Summed feature 3 consists of C15\\xc3\\xa2\\xc2\\x80\\xc2\\x8a:\\xc3\\xa2\\xc2\\x80\\xc2\\x8a0 iso 2-OH and/or C16\\xc3\\xa2\\xc2\\x80\\xc2\\x8a:\\xc3\\xa2\\xc2\\x80\\xc2\\x8a1\\xc3\\x8f\\xc2\\x897\\xc3\\x8f\\xc2\\xb2; summed feature 7 consists of C19\\xc3\\xa2\\xc2\\x80\\xc2\\x8a:\\xc3\\xa2\\xc2\\x80\\xc2\\x8a0cyclo \\xc3\\x8f\\xc2\\x899\\xc3\\x8f\\xc2\\xb2,/C19\\xc3\\xa2\\xc2\\x80\\xc2\\x8a:\\xc3\\xa2\\xc2\\x80\\xc2\\x8a0cyclo \\xc3\\x8f\\xc2\\x8910\\xc3\\x8f\\xc2\\xb2,/C19\\xc3\\xa2\\xc2\\x80\\xc2\\x8a:\\xc3\\xa2\\xc2\\x80\\xc2\\x8a1\\xc3\\x8f\\xc2\\x896\\xc3\\x8f\\xc2\\xb2 and/or an unknown compound at a retention time of 18.85\\xc3\\xa2\\xc2\\x80\\xc2\\x89min\\nThe fermentation profiles of WILCCON 0076T corresponded with the presence or absence of the respective metabolic genes or pathways in the annotated draft genome. The presence of one putative 1-phosphofructokinase gene (pfkB; LB941_07595) and two putative fructose-bisphosphate aldolase genes (fbaA; LB941_03010 and LB941_06780) of the Embden\\xc3\\xa2\\xc2\\x80\\xc2\\x93Meyerhoff pathway and one putative pyruvate formate lyase gene (pflA; LB941_11365), as well as the absence of mannitol dehydrogenase (mdh), lactaldehyde dehydrogenase (aldA) and glycerol/diol dehydratase (pduCDE) genes matched the typical gene contents of homofermentative lactobacilli [31]. Interestingly, the bifunctional acetaldehyde-CoA/alcohol dehydrogenase gene (adhE; LB941_00530) known to be present occasionally in homofermentative lactobacilli was also found in this strain. Its existence may be linked to pyruvate formate lyase which generates acetyl-CoA [31]. Our in silico prediction of the homofermentative nature of WILCCON 0076T was validated via a glucose fermentation assay using an inverted Durham tube to detect the presence (heterofermentative) or absence (homofermentative) of gas production.\\nThe presence of a complete motility locus (scaffolds 3 and 26) consisting of 44 genes of the fli, flg, che and mot operons involved in flagellar assembly, regulation and chemotaxis in the annotated draft genome of WILCCON 0076T is not surprising given its close phylogenetic relationship to other motile species of the former L. salivarius clade (Fig. S3a and Table S3). This in silico prediction of motility was supported by the detection of flagellin (FliC; MCP0886209), flagellar filament capping protein (FliD; MCP0886216) and flagellar hook-associated protein 1 (FlgK; MCP0886219) in the culture supernatant of WILCCON 0076T, which substantiated the functional expression and extracellular localization of these motility-related proteins (Fig. S3b). The apparent motility of WILCCON 0076T was validated in vitro by a soft agar motility assay. WILCCON 0076T was observed to disperse outward from the initial inoculation spots (of 2.5\\xc3\\xa2\\xc2\\x80\\xc2\\x89\\xc3\\x82\\xc2\\xb5l overnight cultures) on 0.4\\xc3\\xa2\\xc2\\x80\\xc2\\x8a% (w/v) soft agar MRS plates (supplemented with 0.05\\xc3\\xa2\\xc2\\x80\\xc2\\x89g l\\xc3\\xa2\\xc2\\x88\\xc2\\x921 2,3,5-triphenyl tetrazolium chloride to stain metabolically active cells) after 1\\xc3\\xa2\\xc2\\x80\\xc2\\x89day of incubation at 37\\xc3\\xa2\\xc2\\x80\\xc2\\x89\\xc3\\x82\\xc2\\xb0C under anoxic conditions (Fig. S3c).\\nIn summary, several lines of evidence from the chemotaxonomic and phenotypic characterization indicated clear physiological and morphological differences between WILCCON 0076T and all so far described species of the genus Ligilactobacillus including the most closely related type strain, Lg. agilis DSM 20509T. Consistent with the phylogenetic analyses presented above, WILCCON 0076T cannot be assigned to any existing species of the genus Ligilactobacillus and thus, represents a novel species of this genus for which we propose the name Ligilactobacillus ubinensis sp. nov.\\n\\xc3\\x82\\xc2\\xa0'\n",
      "b'Physiology and chemotaxonomy\\xc3\\x82\\xc2\\xa0'\n",
      "b'Description of Ligilactobacillus ubinensis sp. nov.\\xc3\\x82\\xc2\\xa0Ligilactobacillus ubinensis (u.bi.nen\\xc3\\xa2\\xc2\\x80\\xc2\\x99.sis. L. adj. ubinensis pertaining to the island of Ubin (Pulau Ubin, Singapore) from where the type strain was isolated).\\nCells are Gram-stain-positive, non-spore-forming, catalase-negative, urease-negative and rod-shaped with dimensions of 2.01\\xc3\\xa2\\xc2\\x80\\xc2\\x937.81\\xc3\\x83\\xc2\\x970.51\\xc3\\xa2\\xc2\\x80\\xc2\\x931.03\\xc3\\xa2\\xc2\\x80\\xc2\\x89\\xc3\\x82\\xc2\\xb5m. Cells are facultatively anaerobic, homofermentative, non-haemolytic and motile. Cell growth is observed at temperatures between 20 \\xc3\\x82\\xc2\\xb0C and 45 \\xc3\\x82\\xc2\\xb0C (optimum at 30\\xc3\\xa2\\xc2\\x80\\xc2\\x89\\xc3\\x82\\xc2\\xb0C and 37\\xc3\\xa2\\xc2\\x80\\xc2\\x89\\xc3\\x82\\xc2\\xb0C), pH values of 4 to 8 (optimum at 7 at 30\\xc3\\xa2\\xc2\\x80\\xc2\\x89\\xc3\\x82\\xc2\\xb0C and 37\\xc3\\xa2\\xc2\\x80\\xc2\\x89\\xc3\\x82\\xc2\\xb0C) and with less than 4\\xc3\\xa2\\xc2\\x80\\xc2\\x8a% (w/v) sodium chloride. Growth at pH 4 to 6 is higher at 37\\xc3\\xa2\\xc2\\x80\\xc2\\x89\\xc3\\x82\\xc2\\xb0C and growth at pH 7 to 8 is higher at 30\\xc3\\xa2\\xc2\\x80\\xc2\\x89\\xc3\\x82\\xc2\\xb0C. Colonies on MRS plates appear white, smooth and round with diameters of 1.5\\xc3\\xa2\\xc2\\x80\\xc2\\x932.0\\xc3\\xa2\\xc2\\x80\\xc2\\x89mm after incubation at 30\\xc3\\xa2\\xc2\\x80\\xc2\\x89\\xc3\\x82\\xc2\\xb0C or 37\\xc3\\xa2\\xc2\\x80\\xc2\\x89\\xc3\\x82\\xc2\\xb0C for two\\xc3\\xa2\\xc2\\x80\\xc2\\x89days under anoxic conditions. Under oxic conditions, colonies form only after at least three\\xc3\\xa2\\xc2\\x80\\xc2\\x89days of incubation at 30\\xc3\\xa2\\xc2\\x80\\xc2\\x89\\xc3\\x82\\xc2\\xb0C but not at 37\\xc3\\xa2\\xc2\\x80\\xc2\\x89\\xc3\\x82\\xc2\\xb0C. Gas is not produced from glucose. Both d-lactic acid and l-lactic acid are produced at a ratio of 1\\xc3\\xa2\\xc2\\x80\\xc2\\x8a:\\xc3\\xa2\\xc2\\x80\\xc2\\x8a20 as the end products of glucose fermentation. In the API 50 CHL test system (incubated at 37\\xc3\\xa2\\xc2\\x80\\xc2\\x89\\xc3\\x82\\xc2\\xb0C for 48\\xc3\\xa2\\xc2\\x80\\xc2\\x89h), acids are produced from d-xylose, d-galactose, d-glucose, d-fructose, d-mannose, N-acetylglucosamine, amygdalin, arbutin, aesculin, salicin, cellobiose, maltose, lactose, melibiose, sucrose, trehalose, raffinose and gentiobiose, but not from glycerol, erythritol, d-arabinose, l-arabinose, d-ribose, l-xylose, d-adonitol, methyl \\xc3\\x8e\\xc2\\xb2-d-xylopyranoside, l-sorbose, l-rhamnose, dulcitol, inositol, d-mannitol, d-sorbitol, methyl \\xc3\\x8e\\xc2\\xb1-d-mannopyranoside, methyl \\xc3\\x8e\\xc2\\xb1-d-glucopyranoside, inulin, melezitose, starch, glycogen, xylitol, turanose, d-lyxose, d-tagatose, d-fucose, l-fucose, d-arabitol, l-arabitol, potassium gluconate, potassium 2-ketogluconate and potassium 5-ketogluconate. The major cellular fatty acids are C16\\xc3\\xa2\\xc2\\x80\\xc2\\x8a:\\xc3\\xa2\\xc2\\x80\\xc2\\x8a0, C18\\xc3\\xa2\\xc2\\x80\\xc2\\x8a:\\xc3\\xa2\\xc2\\x80\\xc2\\x8a1\\xc3\\x8f\\xc2\\x899\\xc3\\x8f\\xc2\\xb2, C18\\xc3\\xa2\\xc2\\x80\\xc2\\x8a:\\xc3\\xa2\\xc2\\x80\\xc2\\x8a1\\xc3\\x8f\\xc2\\x897\\xc3\\x8f\\xc2\\xb2 and summed feature 7. The cell-wall peptidoglycan type is A1\\xc3\\x8e\\xc2\\xb3 (meso-Dpm-direct).\\nThe type strain is WILCCON 0076T (=DSM 114293T=LMG 32698T), isolated from the wild ferment of pieces of a fallen durian fruit collected on the island of Ubin (Pulau Ubin), Singapore. The draft genome of the type strain is 2\\xc3\\xa2\\xc2\\x80\\xc2\\x8a589\\xc3\\xa2\\xc2\\x80\\xc2\\x8a491\\xc3\\xa2\\xc2\\x80\\xc2\\x89bp in size with a DNA G+C content of 33.7\\xc3\\xa2\\xc2\\x80\\xc2\\x89mol%. The GenBank/ENA/DDBJ accession numbers of the 16S rRNA gene sequence, the whole genome shotgun project, and the draft genome assembly of strain WILCCON 0076T are OK166812, JAIULA010000000 and GCA_024160875, respectively.\\n\\xc3\\x82\\xc2\\xa0'\n",
      "b'Description of Ligilactobacillus ubinensis sp. nov.\\xc3\\x82\\xc2\\xa0'\n",
      "b'Funding information\\xc3\\x82\\xc2\\xa0This study was funded by Wilmar International Limited (WIL@NUS Corporate Laboratory, Singapore).\\n\\xc3\\x82\\xc2\\xa0'\n",
      "b'Funding information\\xc3\\x82\\xc2\\xa0'\n",
      "b'Acknowledgements\\xc3\\x82\\xc2\\xa0Sample collection and strain deposition were kindly approved by National Parks Board (NParks), Singapore under the permit NP/RP20-124a. We appreciate the support from Maxim Sheludchenko (Wilmar International Limited, Singapore) on endospore staining as well as from Annie Zhi Qing Sim (Wilmar International Limited, Singapore) on de novo protein sequencing. We further thank Matthew Wook Chang and In Young Hwang [Synthetic Biology for Clinical and Technological Innovation (SynCTI), National University of Singapore, Singapore] for providing reference strains Streptococcus pyogenes DSM 20565T and Streptococcus salivarius subsp. thermophilus DSM 20617T for the haemolysis assay.\\n\\xc3\\x82\\xc2\\xa0'\n",
      "b'Acknowledgements\\xc3\\x82\\xc2\\xa0'\n",
      "b'Author contribution\\xc3\\x82\\xc2\\xa0Y.C.H.: Conceptualization, Data curation, Formal Analysis, Investigation, Visualization, Writing \\xc3\\xa2\\xc2\\x80\\xc2\\x93 original draft, Writing \\xc3\\xa2\\xc2\\x80\\xc2\\x93 review & editing. N.M., B.C., B.Z.L.L., A.C.H.L. and S.S.: Investigation, Writing \\xc3\\xa2\\xc2\\x80\\xc2\\x93 review & editing. W.J.G.W.: Writing \\xc3\\xa2\\xc2\\x80\\xc2\\x93 review & editing. S.K.: Conceptualization, Project administration, Supervision, Writing \\xc3\\xa2\\xc2\\x80\\xc2\\x93 original draft, Writing \\xc3\\xa2\\xc2\\x80\\xc2\\x93 review & editing.\\n\\xc3\\x82\\xc2\\xa0'\n",
      "b'Author contribution\\xc3\\x82\\xc2\\xa0'\n",
      "b'Conflicts of interest\\xc3\\x82\\xc2\\xa0All authors were employed by Wilmar International Limited, Singapore. This does not alter the authors\\xc3\\xa2\\xc2\\x80\\xc2\\x99 adherence to IJSEM policies on sharing data and materials.\\n\\xc3\\x82\\xc2\\xa0'\n",
      "b'Conflicts of interest\\xc3\\x82\\xc2\\xa0'\n",
      "b'References\\xc3\\x82\\xc2\\xa0Zheng J, Wittouck S, Salvetti E, Franz CMAP, Harris HMB, et al. A taxonomic note on the genus Lactobacillus: description of 23 novel genera, emended description of the genus Lactobacillus Beijerinck 1901, and union of Lactobacillaceae and Leuconostocaceae. \\xc3\\x82\\xc2\\xa0Int J Syst Evol Microbiol. 2020;70:2782\\xc3\\xa2\\xc2\\x80\\xc2\\x93858.\\xc3\\x82\\xc2\\xa0 doi: http://dx.doi.org/10.1099/ijsem.0.004107. PubMed PMID: 32293557.Duar RM, Lin XB, Zheng J, Martino ME, Grenier T, et al. Lifestyles in transition: evolution and natural history of the genus Lactobacillus. \\xc3\\x82\\xc2\\xa0FEMS Microbiol Rev. 2017;41:S27\\xc3\\xa2\\xc2\\x80\\xc2\\x9348.\\xc3\\x82\\xc2\\xa0 doi: http://dx.doi.org/10.1093/femsre/fux030. PubMed PMID: 28673043.Zhang S, Oh J-H, Alexander LM, \\xc3\\x83\\xc2\\x96z\\xc3\\x83\\xc2\\xa7am M, van Pijkeren J-P. d-Alanyl-d-alanine ligase as a broad-host-range counterselection marker in vancomycin-resistant lactic acid bacteria. \\xc3\\x82\\xc2\\xa0J Bacteriol. 2018;200e00607-17\\xc3\\x82\\xc2\\xa0 doi: http://dx.doi.org/10.1128/JB.00607-17.Neville BA, Forde BM, Claesson MJ, Darby T, Coghlan A, et al. Characterization of pro-inflammatory flagellin proteins produced by Lactobacillus ruminis and related motile Lactobacilli. \\xc3\\x82\\xc2\\xa0PLoS One. 2012;7e40592\\xc3\\x82\\xc2\\xa0 doi: http://dx.doi.org/10.1371/journal.pone.0040592.Cousin FJ, Lynch SM, Harris HMB, McCann A, Lynch DB, et al. Detection and genomic characterization of motility in Lactobacillus curvatus: confirmation of motility in a species outside the Lactobacillus salivarius clade. \\xc3\\x82\\xc2\\xa0Appl Environ Microbiol. 2015;81:1297\\xc3\\xa2\\xc2\\x80\\xc2\\x93308.\\xc3\\x82\\xc2\\xa0 doi: http://dx.doi.org/10.1128/AEM.03594-14. PubMed PMID: 25501479.Tallant T, Deb A, Kar N, Lupica J, de Veer MJ, et al. Flagellin acting via TLR5 is the major activator of key signaling pathways leading to NF-\\xc3\\x8e\\xc2\\xbaB and proinflammatory gene program activation in intestinal epithelial cells. \\xc3\\x82\\xc2\\xa0BMC Microbiol. 2004;433\\xc3\\x82\\xc2\\xa0 doi: http://dx.doi.org/10.1186/1471-2180-4-33.Harris HMB, Bourin MJB, Claesson MJ, O\\xc3\\xa2\\xc2\\x80\\xc2\\x99Toole PW. Phylogenomics and comparative genomics of Lactobacillus salivarius, a mammalian gut commensal. \\xc3\\x82\\xc2\\xa0Microb Genom. 2017;3e000115\\xc3\\x82\\xc2\\xa0 doi: http://dx.doi.org/10.1099/mgen.0.000115.Bolger AM, Lohse M, Usadel B. Trimmomatic: a flexible trimmer for Illumina sequence data. \\xc3\\x82\\xc2\\xa0Bioinformatics. 2014;30:2114\\xc3\\xa2\\xc2\\x80\\xc2\\x9320.\\xc3\\x82\\xc2\\xa0 doi: http://dx.doi.org/10.1093/bioinformatics/btu170. PubMed PMID: 24695404.Prjibelski A, Antipov D, Meleshko D, Lapidus A, Korobeynikov A. Using SPAdes de novo assembler. \\xc3\\x82\\xc2\\xa0Curr Protoc Bioinformatics. 2020;70e102\\xc3\\x82\\xc2\\xa0 doi: http://dx.doi.org/10.1002/cpbi.102.Parks DH, Imelfort M, Skennerton CT, Hugenholtz P, Tyson GW. CheckM: assessing the quality of microbial genomes recovered from isolates, single cells, and metagenomes. \\xc3\\x82\\xc2\\xa0Genome Res. 2015;25:1043\\xc3\\xa2\\xc2\\x80\\xc2\\x9355.\\xc3\\x82\\xc2\\xa0 doi: http://dx.doi.org/10.1101/gr.186072.114. PubMed PMID: 25977477.Seemann T. Prokka: rapid prokaryotic genome annotation. \\xc3\\x82\\xc2\\xa0Bioinformatics. 2014;30:2068\\xc3\\xa2\\xc2\\x80\\xc2\\x939.\\xc3\\x82\\xc2\\xa0 doi: http://dx.doi.org/10.1093/bioinformatics/btu153. PubMed PMID: 24642063.Kanehisa M, Goto S. KEGG: Kyoto Encyclopedia of Genes and Genomes. \\xc3\\x82\\xc2\\xa0Nucleic Acids Res. 2000;28:27\\xc3\\xa2\\xc2\\x80\\xc2\\x9330.\\xc3\\x82\\xc2\\xa0 doi: http://dx.doi.org/10.1093/nar/28.1.27.Agarwala R, Barrett T, Beck J, Benson DA, Bollin C. Database resources of the National Center for Biotechnology Information. \\xc3\\x82\\xc2\\xa0Nucleic Acids Res. 2018;46:D8\\xc3\\xa2\\xc2\\x80\\xc2\\x93D13.\\xc3\\x82\\xc2\\xa0 doi: http://dx.doi.org/10.1093/nar/gkx1095.Meier-Kolthoff JP, G\\xc3\\x83\\xc2\\xb6ker M, Spr\\xc3\\x83\\xc2\\xb6er C, Klenk HP. When should a DDH experiment be mandatory in microbial taxonomy? \\xc3\\x82\\xc2\\xa0Arch Microbiol. 2013;195:413\\xc3\\xa2\\xc2\\x80\\xc2\\x938.\\xc3\\x82\\xc2\\xa0 doi: http://dx.doi.org/10.1007/s00203-013-0888-4. PubMed PMID: 23591456.Meier-Kolthoff JP, Auch AF, Klenk HP, G\\xc3\\x83\\xc2\\xb6ker M. Genome sequence-based species delimitation with confidence intervals and improved distance functions. \\xc3\\x82\\xc2\\xa0BMC Bioinformatics. 2013;1460\\xc3\\x82\\xc2\\xa0 doi: http://dx.doi.org/10.1186/1471-2105-14-60. PubMed PMID: 23432962.Edgar RC. MUSCLE: multiple sequence alignment with high accuracy and high throughput. \\xc3\\x82\\xc2\\xa0Nucleic Acids Res. 2004;32:1792\\xc3\\xa2\\xc2\\x80\\xc2\\x937.\\xc3\\x82\\xc2\\xa0 doi: http://dx.doi.org/10.1093/nar/gkh340. PubMed PMID: 15034147.Madeira F, Park YM, Lee J, Buso N, Gur T, et al. The EMBL-EBI search and sequence analysis tools APIs in 2019. \\xc3\\x82\\xc2\\xa0Nucleic Acids Res. 2019;47:W636\\xc3\\xa2\\xc2\\x80\\xc2\\x9341.\\xc3\\x82\\xc2\\xa0 doi: http://dx.doi.org/10.1093/nar/gkz268. PubMed PMID: 30976793.Nguyen L-T, Schmidt HA, von Haeseler A, Minh BQ. IQ-TREE: a fast and effective stochastic algorithm for estimating maximum-likelihood phylogenies. \\xc3\\x82\\xc2\\xa0Mol Biol Evol. 2015;32:268\\xc3\\xa2\\xc2\\x80\\xc2\\x9374.\\xc3\\x82\\xc2\\xa0 doi: http://dx.doi.org/10.1093/molbev/msu300. PubMed PMID: 25371430.Kalyaanamoorthy S, Minh BQ, Wong TKF, von Haeseler A, Jermiin LS. ModelFinder: fast model selection for accurate phylogenetic estimates. \\xc3\\x82\\xc2\\xa0Nat Methods. 2017;14:587\\xc3\\xa2\\xc2\\x80\\xc2\\x939.\\xc3\\x82\\xc2\\xa0 doi: http://dx.doi.org/10.1038/nmeth.4285. PubMed PMID: 28481363.Hoang DT, Chernomor O, von Haeseler A, Minh BQ, Vinh LS. UFBoot2: improving the ultrafast bootstrap approximation. \\xc3\\x82\\xc2\\xa0Mol Biol Evol. 2018;35:518\\xc3\\xa2\\xc2\\x80\\xc2\\x9322.\\xc3\\x82\\xc2\\xa0 doi: http://dx.doi.org/10.1093/molbev/msx281.Pritchard L, Glover RH, Humphris S, Elphinstone JG, Toth IK. Genomics and taxonomy in diagnostics for food security: soft-rotting enterobacterial plant pathogens. \\xc3\\x82\\xc2\\xa0Anal Methods. 2016;8:12\\xc3\\xa2\\xc2\\x80\\xc2\\x9324.\\xc3\\x82\\xc2\\xa0 doi: http://dx.doi.org/10.1039/C5AY02550H.Wittouck S, Wuyts S, Meehan CJ, van Noort V, Lebeer S. A genome-based species taxonomy of the Lactobacillus\\xc3\\x82\\xc2\\xa0genus complex. \\xc3\\x82\\xc2\\xa0mSystems. 2019;4e00264-19\\xc3\\x82\\xc2\\xa0 doi: http://dx.doi.org/10.1128/mSystems.00264-19.Emms DM, Kelly S. OrthoFinder: phylogenetic orthology inference for comparative genomics. \\xc3\\x82\\xc2\\xa0Genome Biol. 2019;20:238.\\xc3\\x82\\xc2\\xa0 doi: http://dx.doi.org/10.1186/s13059-019-1832-y. PubMed PMID: 31727128.Eddy SR. Accelerated profile HMM searches. \\xc3\\x82\\xc2\\xa0PLoS Comput Biol. 2011;7e1002195\\xc3\\x82\\xc2\\xa0 doi: http://dx.doi.org/10.1371/journal.pcbi.1002195.Katoh K, Standley DM. MAFFT multiple sequence alignment software version 7: improvements in performance and usability. \\xc3\\x82\\xc2\\xa0Mol Biol Evol. 2013;30:772\\xc3\\xa2\\xc2\\x80\\xc2\\x9380.\\xc3\\x82\\xc2\\xa0 doi: http://dx.doi.org/10.1093/molbev/mst010. PubMed PMID: 23329690.Capella-Guti\\xc3\\x83\\xc2\\xa9rrez S, Silla-Mart\\xc3\\x83\\xc2\\xadnez JM, Gabald\\xc3\\x83\\xc2\\xb3n T. trimAl: a tool for automated alignment trimming in large-scale phylogenetic analyses. \\xc3\\x82\\xc2\\xa0Bioinformatics. 2009;25:1972\\xc3\\xa2\\xc2\\x80\\xc2\\x933.\\xc3\\x82\\xc2\\xa0 doi: http://dx.doi.org/10.1093/bioinformatics/btp348. PubMed PMID: 19505945.Sarachu M, Colet M. wEMBOSS: a web interface for EMBOSS. \\xc3\\x82\\xc2\\xa0Bioinformatics. 2005;21:540\\xc3\\xa2\\xc2\\x80\\xc2\\x931.\\xc3\\x82\\xc2\\xa0 doi: http://dx.doi.org/10.1093/bioinformatics/bti031. PubMed PMID: 15388516.Kim M, Oh HS, Park SC, Chun J. Towards a taxonomic coherence between average nucleotide identity and 16S rRNA gene sequence similarity for species demarcation of prokaryotes. \\xc3\\x82\\xc2\\xa0Int J Syst Evol Microbiol. 2014;64:346\\xc3\\xa2\\xc2\\x80\\xc2\\x9351.\\xc3\\x82\\xc2\\xa0 doi: http://dx.doi.org/10.1099/ijs.0.059774-0. PubMed PMID: 24505072.Weiss N, Schillinger U, Laternser M, Kandler O. Lactobacillus sharpeae sp.nov. and Lactobacillus agilis sp.nov., two new species of homofermentative, meso-diaminopimelic acid-containing lactobacilli isolated from sewage. \\xc3\\x82\\xc2\\xa0Zbl Bakt Hyg, I Abt Orig C. 1981;2:242\\xc3\\xa2\\xc2\\x80\\xc2\\x9353.\\xc3\\x82\\xc2\\xa0 doi: http://dx.doi.org/10.1016/S0721-9571(81)80005-1.Schumann P. \\xc3\\x82\\xc2\\xa0\\xc3\\x82\\xc2\\xa0Methods in Microbiology. Academic Press. 38. 2011. p. 101-29. In vol ppZheng J, Ruan L, Sun M, G\\xc3\\x83\\xc2\\xa4nzle M. A genomic view of lactobacilli and pediococci demonstrates that phylogeny matches ecology and physiology. \\xc3\\x82\\xc2\\xa0Appl Environ Microbiol. 2015;81:7233\\xc3\\xa2\\xc2\\x80\\xc2\\x9343.\\xc3\\x82\\xc2\\xa0 doi: http://dx.doi.org/10.1128/AEM.02116-15.\\xc3\\x82\\xc2\\xa0'\n",
      "b'References\\xc3\\x82\\xc2\\xa0'\n"
     ]
    }
   ],
   "source": [
    "for section in article:\n",
    "   print(section.text.encode('utf8'))\n",
    "   "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "44e7ec66-593a-4bc3-af67-287072172bbd",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "b'Introduction\\xc3\\x82\\xc2\\xa0'"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "article[7].text.encode(\"utf-8\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "d753f6ba-249f-40e7-a7d0-02a71d07918a",
   "metadata": {},
   "outputs": [
    {
     "ename": "TypeError",
     "evalue": "'NoneType' object does not support item assignment",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mTypeError\u001b[0m                                 Traceback (most recent call last)",
      "Cell \u001b[1;32mIn [12], line 4\u001b[0m\n\u001b[0;32m      1\u001b[0m example_element \u001b[38;5;241m=\u001b[39m soup\u001b[38;5;241m.\u001b[39mfind(\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mdiv\u001b[39m\u001b[38;5;124m'\u001b[39m, {\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mclass\u001b[39m\u001b[38;5;124m'\u001b[39m: \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mexample-class\u001b[39m\u001b[38;5;124m'\u001b[39m})\n\u001b[0;32m      3\u001b[0m \u001b[38;5;66;03m# Remove the class from the element\u001b[39;00m\n\u001b[1;32m----> 4\u001b[0m \u001b[43mexample_element\u001b[49m\u001b[43m[\u001b[49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[38;5;124;43mclass\u001b[39;49m\u001b[38;5;124;43m'\u001b[39;49m\u001b[43m]\u001b[49m \u001b[38;5;241m=\u001b[39m []\n",
      "\u001b[1;31mTypeError\u001b[0m: 'NoneType' object does not support item assignment"
     ]
    }
   ],
   "source": [
    "example_element = soup.find('div', {'class': 'example-class'})\n",
    "\n",
    "# Remove the class from the element\n",
    "example_element['class'] = []\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "009e5231-cfc5-4f2e-b722-37658ea0a393",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "910897d0-a521-454b-8570-7019dde369da",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'-B[x\\x14Ez'"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "bcb3c634-2d23-4c4c-9bac-59fc2aff97bb",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
